xbmc/interfaces/legacy/AddonUtils.h

```diff
@@ -68,6 +69,9 @@ namespace XBMCAddonUtils
   };
 
 #define LOCKGUI XBMCAddonUtils::GuiLock __gl
+#define LOCKGUIIF(cond) std::unique_ptr<XBMCAddonUtils::GuiLock> __gl; \
```

Comment:
I really hate those crappy marcros. sorry, I couldn't resist :)
The problem here is that this crappy marco does not only was its name suggests. It also exits the GIL.